src/connection/index.js

```diff
@@ -325,13 +319,13 @@ class ConnectionFSM extends BaseConnection {
           return this.close(maybeUnexpectedEnd(err))
         }
 
-        const conn = observeConnection(null, this.switch.crypto.tag, _conn, this.switch.observer)
-
-        this.conn = this.switch.crypto.encrypt(this.ourPeerInfo.id, conn, this.theirPeerInfo.id, (err) => {
```

Comment:
I found an edge case here, where if the connection was terminated while encryption was being negotiated, `this.conn` would get resolved and would later throw an error when secio went to set the inner connection. This avoids that.

Reply:
Nice catch!